extensions/classifiers/SVMPredictionService.js

```diff
@@ -26,109 +26,222 @@
+        }
+
+        // Find kernel values for supportVectors and given input. Assumes that
+        // input has same dimension and data type as any of the supportVectors.
+        var kvalues = this.kernel(kernelParams, supportVectors, input);
+
+        var votes = [];
+        for (var i = 0; i < classes.length; i++) {
+          votes.push(0);
+        }
+
+        var pairwiseProb = [];
+        for (var i = 0; i < classes.length; i++) {
+          pairwiseProb.push([]);
+          for (var j = 0; j < classes.length; j++) {
+            pairwiseProb[i].push(0);
+          }
+        }
+
+        var p = 0;
+        for (var i = 0; i < classes.length; i++) {
+          for (var j = i + 1; j < classes.length; j++) {
+            var si = startIndices[i];
+            var sj = startIndices[j];
+            var ci = nSupport[i];
+            var cj = nSupport[j];
+            var minProb = 1e-7;
+
+            var coef1 = dualCoef[j - 1];
+            var coef2 = dualCoef[i];
+
+            var sum = 0;
+            for (var k = 0; k < ci; k++) {
+              sum += kvalues[si + k] * coef1[si + k];
+            }
+
+            for (var k = 0; k < cj; k++) {
+              sum += kvalues[sj + k] * coef2[sj + k];
+            }
+
+            // NOTE: libsvm substracts the intercept from sum in its prediction
+            // function. Here intercept is added because sci-kit negates the
+            // intercept before passing it on to libsvm for prediction.
+            // For more info see github following issue:
+            // https://github.com/oppia/oppia/issues/4166
+            sum += intercept[p];
+
+            // The following approach to calculate pairwise probabilities was
+            // proposed by platt. For more info on LibSVM's implementation
+            // of platt scaling, read following paper:
+            // https://www.csie.ntu.edu.tw/~cjlin/papers/plattprob.pdf
+            // Also take a look at following implementation by LibSVM:
+            // https://github.com/arnaudsj/libsvm/blob/master/svm.cpp#L2552
+            var f = probA[p] * sum + probB[p];
+            var prob = 0;
+            if (f >= 0) {
+              prob = Math.exp(-f) / (1 + Math.exp(-f));
+            } else {
+              prob = 1 / (1 + Math.exp(f));
+            }
+            prob = Math.min(Math.max(prob, minProb), 1 - minProb);
+            pairwiseProb[i][j] = prob;
+            pairwiseProb[j][i] = 1 - prob;
+            p++;
+          }
+        }
+
+        // Find multiclass probabilities.
+        // NOTE: This function is implemented as it is given in LibSVM.
+        // For more information on exact approach used, read following paper:
+        // https://www.csie.ntu.edu.tw/~cjlin/papers/svmprob/svmprob.pdf
+        // Also take a look at implementation by LibSVM:
+        // https://github.com/arnaudsj/libsvm/blob/master/svm.cpp#L1829
+        probs = this.multiclassProbability(classes.length, pairwiseProb);
```

Comment:
Since multiclassProbability is a function, use a name that starts with a verb. E.g. calculateMulticlassProbabilities or something like that.

Also the var name "probs" indicates a plural whereas "probability" is singular. Let's be consistent.